steward/src/allocation.rs

```diff
@@ -181,3 +205,44 @@ pub fn to_allocation(
             .collect(),
     }
 }
+
+pub async fn direct_rebalance(
+    cellar_address: H160,
+    tick_weight: Vec<TickWeight>,
+) -> Result<(), Error> {
+    let mut tick_info: Vec<CellarTickInfo> = Vec::new();
+    let config = APP.config();
+    let keystore = path::Path::new(&config.keys.keystore);
+    let keystore = FsKeyStore::create_or_open(keystore).expect("Could not open keystore");
+
+    let name = &config
+        .keys
+        .rebalancer_key
+        .parse()
+        .expect("Could not parse name");
+    let key = keystore.load(name).expect("Could not load key");
+
+    let key = key
+        .to_pem()
+        .parse::<k256::elliptic_curve::SecretKey<k256::Secp256k1>>()
+        .expect("Could not parse key");
+
+    let wallet: LocalWallet = Wallet::from(key);
+
+    let eth_host = config.ethereum.rpc.clone();
+    let client = Provider::<Http>::try_from(eth_host.clone()).unwrap();
+    let client = SignerMiddleware::new(client, wallet.clone());
```

Comment:
In my experience I've run into errors unless I set the chain ID on the wallet. An example from one of the steward commands:

https://github.com/PeggyJV/steward/blob/5948dadad915b4bfa564fd7860dcf5f98d4e9832/steward/src/commands/eth_to_cosmos.rs#L49-L59

Reply:
Fixed with this commit https://github.com/PeggyJV/steward/pull/55/commits/8c72dbd33536f6e42fb57c7bc8d78d03d1dc4dd0

Reply:
The update retrieves the chain ID, but you still need to create a cloned wallet associated with that chain ID as such:

```
 let eth_client = SignerMiddleware::new( 
     provider, 
     ethereum_wallet.clone().with_chain_id(chain_id), 
 ); 
```